com.ibm.streamsx.topology/opt/python/packages/streamsx/scripts/sc.py

```diff
@@ -51,45 +52,98 @@ def _parse_dependencies():
             deps[name] = version
     return deps
 
-def _add_local_toolkits(toolkits, dependencies, topo):
+def _add_local_toolkits(toolkit_paths, dependencies, topo, verify_arg):
     """ A helper function that given the local toolkit paths, and the dependencies (ie toolkits that the apps depends on),
-    checks and adds the local toolkit if it is one of the dependencies
+    finds the latest dependency w/ correct version (could be local or on the buildserver) and checks whether it exists locally, and if so adds it.
+    If the latest dependency w/ correct version is located on the buildserver, don't need to do anything
 
     Arguments:
-        toolkits {List} -- A list of local toolkit directories, each could be a toolkit directory (has info.xml directly inside),
+        toolkit_paths {List} -- A list of local toolkit directories, each could be a toolkit directory (has info.xml directly inside),
                 or a directory consisting of toolkits (no info.xml directly inside)
-        dependencies {Dictionary} -- A dictionary where the key is the names of the dependencies, and the value is its version.
+        dependencies {Dictionary} -- A dictionary consisting of elements where the key is the name of the dependency, and the value is its version.
         topo {topology Object} -- [description]
+        verify_arg {type} -- Disable SSL verification.
     """
+    build_server = BuildService.of_endpoint(verify=verify_arg)
+
+    local_toolkits = _get_all_local_toolkits(toolkit_paths)
+    remote_toolkits = build_server.get_toolkits()
+
+    all_toolkits = local_toolkits + remote_toolkits
+
+    # Iterate through all dependencies (toolkits) needed to build sab file, and check if the dependency already exist on the build server
+    for dependency_name, dependency_version in dependencies.items():
+
+        # All the toolkits that match dependency_name regardless of version
+        matching_toolkits = [toolkit for toolkit in all_toolkits if toolkit.name == dependency_name]
+
+        if not matching_toolkits:
+            # No toolkits exists on the remote buildserver or locally that can satsify this dependency, continue and let SPL compiler throw error
+            continue
+
+        # Toolkit with the highest verion that still satisfies the dependency_version
+        latest_compatible_toolkit = matching_toolkits[0]
```

Comment:
Doesn't `latest_compatible_toolkit` need to start out as `None`?

`matching_toolkits[0]` is a match on name only, it may not satisfy the dependency, so if it is `5.0.0` but the dependency is `[1.0.0,2.0.0]` then it will be selected.